cmd/shortener/config/flags.go

```diff
@@ -0,0 +1,53 @@
+package config
+
+import (
+    "flag"
+    "os"
+    "strings"
+    "fmt"
+    "strconv"
+)
+
+func splitHostPort(addr string) (string, int, error) {
+	parts := strings.Split(addr, ":")
+	if len(parts) != 2 {
+		return "", 0, fmt.Errorf("invalid address format")
+	}
+    num, err := strconv.Atoi(parts[1])
+	if err != nil {
+		fmt.Println("can not str into number:", err)
+		return parts[0], 0, nil
+	}
+	return parts[0], num, nil
+}
+
+func ParseFlags() Settings {
```

Comment:
Тут можно использовать указатель `*Settings` во избежание лишнего копирвоания.  
```suggestion
func ParseFlags() *Settings {
```